server/utils/mailer.js

```diff
@@ -0,0 +1,87 @@
+const { isObject } = require('lodash');
+const nodemailer = require('nodemailer');
+const smtpTransport = require('nodemailer-smtp-transport');
+const { promisify } = require('util');
+
+const config = require('../../config');
+
+const logger = config.logger('mailer');
+const transporter = createTransport();
+const sendMail = promisify(transporter.sendMail.bind(transporter));
+
+const testMails = [];
+exports.testMails = testMails;
+
+/**
+ * Sends an email using the default Nodemailer transport.
+ *
+ * In test mode, emails are not sent, but are instead stored in the `testEmails`
+ * array, which is exported from this module.
+ *
+ * @async
+ * @function
+ * @param {Object} options - Email properties.
+ * @see {@link https://nodemailer.com/message/}
+ */
+exports.sendMail = async function(options) {
+  if (!isObject(options)) {
+    throw new Error('Mail options must be an object');
+  } else if (!options.to) {
+    throw new Error('Mail `to` option is required');
+  } else if (!options.subject) {
+    throw new Error('Mail `subject` option is required');
+  } else if (!options.text) {
+    throw new Error('Mail `text` option is required');
+  }
+
+  const start = new Date().getTime();
+
+  const email = {
+    from: config.mail.fromName ? `"${config.mail.fromName}" <${config.mail.fromAddress}>` : config.mail.fromAddress,
+    html: options.html,
+    to: options.to,
+    subject: options.subject,
+    text: options.text
+  };
+
+  if (!config.mail.enabled) {
+
+    // Store test emails in memory.
+    if (config.env === 'test') {
```

Comment:
Why checking that the config mail is not enabled? Why not pushing directly in the `testMails` even if the config mail is enabled, considering that the config environment is test?

Reply:
The original writer of this code was not right in the head. I have simplified to this:

* If `config.mail.enabled` is false, no emails are sent, including in the test environment. This is way more logical.
* In the test environment, mails are always stored in memory. This is simpler.